hazelcast/src/main/java/com/hazelcast/sql/impl/type/QueryDataTypeUtils.java

```diff
@@ -54,36 +54,94 @@
  * Utility methods for SQL data types.
```

Comment:
Might be worth mentioning that the descriptions are generated using JOL. I was wondering what they mean, until I've read the PR description. I'm still wondering what they mean, but I trust them more :).